ruby_programming/advanced_ruby/blocks.md

```diff
@@ -310,6 +319,20 @@ a_proc.call("apple")
 # => a: apple --- b:
 ~~~
 
+which is also why this is possible:
+
+~~~ruby
+nested_array = [[1, 2], [3, 4], [5, 6]]
+nested_array.select {|a, b| a + b > 10 }
+
+# => [5, 6]
+~~~
+
+As you can see, `#select` has two arguments specified `|a, b|`, on each iteration we pass a single element of nested_array into the block. On the first iteration this is: `[1, 2]`, this array now, is deconstructed automatically (into a = 1, b = 2) and its values compared as specified. So on to the next rounds of iteration in which we pass `[3,4]`, `[7, 8]`, and `[5, 6]` one by one.
```

Comment:
Sorry to be pedantic but can be make sure the `[3,4]` array has the same spacing as the other ones